### PR TITLE
Config URL Cleanup

### PR DESCRIPTION
**Pull request by mananjadhav**
### Fixed Issues

$ https://github.com/Expensify/App/issues/7489

### Tests
1. Tested that `.env` file changes load the correct environment
2. Verified changes in the Readme
3. Tested App Download link
4. Google Api test done  ~CORS origin from localhost:8080~
5. Plaid Test done ~CORS origin from localhost:8080~

- [x] Verify that no errors appear in the JS console

### QA Steps

**Check correct URL is picked**
1. While PR changes configs and variables, one thing to check is that the App works as expected and all API requests are running to env specific APIs, ie, Staging connects to Staging URL, and so on

**App Download** 
1. Got the settings by clicking on Avatar on LHN
2. Click on `About ->  App Download links` menu
3. Click on `macOS` download link, and it should download the correct bundle

**Plaid Link**
1. Got the settings by clicking on Avatar on LHN
2. Click on `Payments ->  Add Payment` menu
3. Click on "Bank Account" and ensure that Plaid flow is working fine

- [x] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

Screenshot 2022-02-20 at 8 47 51 PM

Screenshot 2022-02-20 at 9 14 03 PM

#### Mobile Web

Screenshot 2022-02-20 at 8 49 55 PM

#### Desktop

Screenshot 2022-02-20 at 8 57 28 PM

#### iOS

Screenshot 2022-02-20 at 8 54 17 PM

#### Android

Screenshot 2022-02-20 at 8 58 30 PM

### CODE REVIEW

**Comment by mananjadhav** [comment]
@rushatgabhane  @iwiznia, three quick questions:

1. How to best test this with .env changes?
2. I've also renamed `EXPENSIFY_URL_COM` to `EXPENSIFY_URL` matching the variable `expensifyUrl`
3. The content variable `CONST.NEWDOT` I am using `NEW_EXPENSIFY_URL`, but note that when on staging it might show Staging URL. I am guessing it should be fine but let me know if you think otherwise.

**Comment by iwiznia** [comment]
> How to best test this with .env changes?

I guess you can change the URLs there and test pointing to staging and prod?

> The content variable CONST.NEWDOT I am using NEW_EXPENSIFY_URL, but note that when on staging it might show Staging URL. I am guessing it should be fine but let me know if you think otherwise.
I think

**Comment by mananjadhav** [comment]
@iwiznia @rushatgabhane Will this go via QA? I am not sure what to add for screenshots and QA steps.

File: .env.production
Comment (rushatgabhane) [inline .env.production:2]: Let's change it to `SECURE_EXPENSIFY_URL` everywhere?

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@mananjadhav looks good, just requesting a tiny change.

File: .env.production
Comment (mananjadhav) [reply .env.production]: UPDATED

**Review by rushatgabhane — APPROVED** [review]
> I guess you can change the URLs there and test pointing to staging and prod?

This works as expected.

@iwiznia LGTM! 🎉️

**Comment by rushatgabhane** [comment]
> Pending: Google Api test, CORS origin from localhost:8080
Pending: Plaid Test, CORS origin from localhost:8080


[This chrome extension](https://chrome.google.com/webstore/detail/cors-unblock/lfhmikememgdcahcdlaciloancbhjino?hl=en) gets the job done.

**Review by iwiznia — APPROVED** [review]
Looks good, but holding merge on:
- The pending tests
- Since this is renaming some widely used variables, you should merge main into this branch right before we merge, to reduce the possibility of some other PR having added usages of the old variables

**Comment by mananjadhav** [comment]
Merged latest `main` and checked the `Plaid` flow. I didn't complete the whole flow, just tested that it is working fine.

**Comment by mvtglobally** [comment]
@mananjadhav @rushatgabhane @iwiznia Anything QA team need to run here ?

**Comment by francoisl** [comment]
@mananjadhav @rushatgabhane @iwiznia Bump on the above please, looks like QA doesn't need to do anything other than check there are overall no errors in the console, can you confirm please?

**Comment by rushatgabhane** [comment]
I'm so sorry about this.
@mananjadhav yes, could you please add the QA steps related to app download, terms page and plaid link.

**Comment by iwiznia** [comment]
I assumed it was no QA because most of it would be caught in regression testing, but if there are specific things to QA here, please let's add them.

**Comment by mananjadhav** [comment]
Updated

**Comment by marcaaron** [comment]
Do we need an `[Action Required]` update or email for this?

**Comment by rushatgabhane** [comment]
> Do we need an [Action Required] update or email for this?

Are you asking if we change the title of PR?

**Comment by marcaaron** [comment]
No, sorry to clarify... if we change the variable names for the environment variables that everyone who works on the codebase might need to know about this change. They may have a bunch of pointless or undefined stuff in their local `.env` file because we changed the names of things.

**Comment by rushatgabhane** [comment]
@marcaaron oh yeah, good call. Here are the renames that need to be made in a` .env` 
It'd best if you post this in an internal slack channel because environment variables aren't used externally.

`EXPENSIFY_URL_CASH` -> `NEW_EXPENSIFY_URL`

`EXPENSIFY_URL_SECURE` -> `SECURE_EXPENSIFY_URL`

`EXPENSIFY_URL_COM` -> `EXPENSIFY_URL`

**Pull request merged**
